test/functional/p2p_compactblocks.py

```diff
@@ -823,12 +827,100 @@ def assert_highbandwidth_states(node, hb_to, hb_from):
         hb_test_node.send_and_ping(msg_sendcmpct(announce=False, version=2))
         assert_highbandwidth_states(self.nodes[0], hb_to=True, hb_from=False)
 
+    def test_compactblock_reconstruction_parallel_reconstruction(self, stalling_peer, delivery_peer, inbound_peer, outbound_peer):
+        """ All p2p connections are inbound except outbound_peer. We test that ultimate parallel slot
+            can only be taken by an outbound node unless prior attempts were done by an outbound
+        """
+        node = self.nodes[0]
+        assert len(self.utxos)
+
+        def announce_cmpct_block(node, peer, txn_count):
+            utxo = self.utxos.pop(0)
+            block = self.build_block_with_transactions(node, utxo, txn_count)
+
+            cmpct_block = HeaderAndShortIDs()
+            cmpct_block.initialize_from_block(block)
+            msg = msg_cmpctblock(cmpct_block.to_p2p())
+            peer.send_and_ping(msg)
+            with p2p_lock:
+                assert "getblocktxn" in peer.last_message
+            return block, cmpct_block
+
+        # First, make delivery_peer, inbound_peer, and outbound_peer high bandwidth
+        block, cmpct_block = announce_cmpct_block(node, delivery_peer, 1)
```

Comment:
Suggestion to simplify it, in case you touch it again:
```diff
index 50cce9318..cd9268f6b 100755
--- a/test/functional/p2p_compactblocks.py
+++ b/test/functional/p2p_compactblocks.py
@@ -847,29 +847,14 @@ class CompactBlocksTest(BitcoinTestFramework):
             return block, cmpct_block
 
         # First, make delivery_peer, inbound_peer, and outbound_peer high 
bandwidth
-        block, cmpct_block = announce_cmpct_block(node, delivery_peer, 1)
-        msg = msg_blocktxn()
-        msg.block_transactions.blockhash = block.sha256
-        msg.block_transactions.transactions = block.vtx[1:]
-        delivery_peer.send_and_ping(msg)
-        assert_equal(int(node.getbestblockhash(), 16), block.sha256)
-        delivery_peer.clear_getblocktxn()
-
-        block, cmpct_block = announce_cmpct_block(node, inbound_peer, 1)
-        msg = msg_blocktxn()
-        msg.block_transactions.blockhash = block.sha256
-        msg.block_transactions.transactions = block.vtx[1:]
-        inbound_peer.send_and_ping(msg)
-        assert_equal(int(node.getbestblockhash(), 16), block.sha256)
-        inbound_peer.clear_getblocktxn()
-
-        block, cmpct_block = announce_cmpct_block(node, outbound_peer, 1)
-        msg = msg_blocktxn()
-        msg.block_transactions.blockhash = block.sha256
-        msg.block_transactions.transactions = block.vtx[1:]
-        outbound_peer.send_and_ping(msg)
-        assert_equal(int(node.getbestblockhash(), 16), block.sha256)
-        outbound_peer.clear_getblocktxn()
+        for peer in [delivery_peer, inbound_peer, outbound_peer]:
+            block, _ = announce_cmpct_block(node, peer, 1)
+            msg = msg_blocktxn()
+            msg.block_transactions.blockhash = block.sha256
+            msg.block_transactions.transactions = block.vtx[1:]
+            peer.send_and_ping(msg)
+            assert_equal(int(node.getbestblockhash(), 16), block.sha256)
+            peer.clear_getblocktxn()
 
         # Test the simple parallel download case...
         for num_missing in [1, 5, 20]:
```

Reply:
done